### PR TITLE
first pass at cleaning out bch related code

### LINKED ISSUE #1: Remove Bitcoin Cash Support
Bitcoin Cash is only superficially similar to Bitcoin. It will also diverge over time, maintaining them both in the same library is a liability.

### PR DESCRIPTION
**Pull request by ProofOfKeags**
this PR addresses #1 

It seems like there may be more things to do related to pulling out BCH related opcodes out of the script interpretation, however, iirc we may be moving the script interpreter out of the package altogether.

### CODE REVIEW

File: bitcoin.cabal [hunk 1]
```diff
@@ -4,7 +4,7 @@ cabal-version: 1.12
 --
 -- see: https://github.com/sol/hpack
 
-name:           haskoin-core
+name:           bitcoin
 version:        0.21.2
```
Comment (GambolingPangolin) [inline bitcoin.cabal]: How do you feel about resetting the version?  I would love to do a few burn-in `0.*` versions, then get to `1.0`.

File: src/Haskoin/Address.hs [hunk 2]
```diff
@@ -5,16 +5,15 @@
...
-Copyright   : No rights reserved
-License     : MIT
-Maintainer  : jprupp@protonmail.ch
-Stability   : experimental
-Portability : POSIX
-
-Base58, CashAddr, Bech32 address and WIF private key serialization support.
--}
+-- |
+--Module      : Haskoin.Address
```
Comment (GambolingPangolin) [inline src/Haskoin/Address.hs]: Does the Haddock come out right when there is no space after the comment token?  (Applies everywhere)

File: src/Haskoin/Address.hs [hunk 3]
```diff
@@ -5,16 +5,15 @@
...
+--Copyright   : No rights reserved
+--License     : MIT
+--Maintainer  : jprupp@protonmail.ch
```
Comment (GambolingPangolin) [inline src/Haskoin/Address.hs]: We prob should just rip out the module level metadata, except for `Stability` and `Portability`, with the goal of setting `Stability` to "stable" eventually.

File: package.yaml [hunk 4]
```diff
@@ -7,7 +7,8 @@ author:
   - Philippe Laprade
   - Jean-Pierre Rupp
   - Matthew Wraith
-maintainer: jprupp@protonmail.ch
+  - Keagan McClelland
+maintainer: keagan.mcclelland@gmail.com
 license: MIT
```
Comment (GambolingPangolin) [inline package.yaml]: I believe we can (and should) change the license to BSD3.

File: src/Haskoin/Address.hs [hunk 5]
```diff
@@ -5,16 +5,15 @@
...
+--Stability   : experimental
+--Portability : POSIX
+--
+--Base58, CashAddr, Bech32 address and WIF private key serialization support.
```
Comment (GambolingPangolin) [inline src/Haskoin/Address.hs]: No `CashAddr` (and below, prob just grep it)

File: src/Haskoin/Address.hs [hunk 6]
```diff
@@ -5,16 +5,15 @@
 {-# LANGUAGE LambdaCase #-}
 {-# LANGUAGE OverloadedStrings #-}
```
Comment (GambolingPangolin) [inline src/Haskoin/Address.hs:6]: How do you feel about adding `LANGUAGE Safe` or `LANGUAGE Trustworthy` where we can?

File: src/Haskoin/Address/Base58.hs [hunk 7]
```diff
@@ -38,84 +36,91 @@ import Haskoin.Crypto.Hash
...
--}
+    where
+        p = isJust . b58' . fromIntegral . ord
+        f = fromMaybe e . b58' . fromIntegral . ord
+        go = listToMaybe $ readInt 58 p f (cs s)
+        e = error "Could not decode base58"
+
+
+-- | Encode an arbitrary 'ByteString' into a its 'Base58' representation,
+-- preserving leading zeroes.
```
Comment (GambolingPangolin) [inline src/Haskoin/Address/Base58.hs:86]: I like this comment style better.  Good call.

File: fourmolu.yaml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+indentation: 4
```
Comment (GambolingPangolin) [inline fourmolu.yaml:1]: _BIKESHEDDING ALERT_ I like 2 spaces

**Comment by GambolingPangolin** [comment]
An easy loose end to fix is getting rid of `SIGHASH_FORKID` (and decendents) and renaming `txSigHashForkId` to `txSigHashSegwitV0` or something.

File: src/Haskoin/Address/Base58.hs (hunk 7 above)
Comment (ProofOfKeags) [reply src/Haskoin/Address/Base58.hs:86]: ha, my formatter did it for me...

File: src/Haskoin/Address.hs (hunk 6 above)
Comment (ProofOfKeags) [reply src/Haskoin/Address.hs:6]: Think it's great, but not needed for this PR. Can we cut a separate ticket for it just to keep things clean?

File: src/Haskoin/Address.hs (hunk 5 above)
Comment (ProofOfKeags) [reply src/Haskoin/Address.hs]: Good catch.

File: package.yaml (hunk 4 above)
Comment (ProofOfKeags) [reply package.yaml]: That doesn't require soliciting permission does it, since MIT is less restrictive?

File: src/Haskoin/Address.hs (hunk 3 above)
Comment (ProofOfKeags) [reply src/Haskoin/Address.hs]: Agreed. Also a separate ticket item. Although I know I invited this by allowing my autoformatter to go hogwild with the reformatting of comments.

File: src/Haskoin/Address.hs (hunk 2 above)
Comment (ProofOfKeags) [reply src/Haskoin/Address.hs]: it does not, I think the transformer on the autoformatter is Wrong™

File: bitcoin.cabal (hunk 1 above)
Comment (ProofOfKeags) [reply bitcoin.cabal]: Completely fine with that.

File: fourmolu.yaml (hunk 8 above)
Comment (ProofOfKeags) [reply fourmolu.yaml:1]: 👎🏼 Booooo. I've been doing 4 spaces my whole Haskell career. I *know* it fucks with let bindings in do blocks. And yet...

@wraithm I know I'm probably gonna get outnumbered here, but maybe break this tie..

**Comment by ProofOfKeags** [comment]
> An easy loose end to fix is getting rid of `SIGHASH_FORKID` (and decendents) and renaming `txSigHashForkId` to `txSigHashSegwitV0` or something.

Good catch. I knew I was gonna miss something cause I didn't comb through all of the code scrutinizing things that don't exist in BTC. I'll remove that here.

File: src/Haskoin/Address.hs (hunk 6 above)
Comment (ProofOfKeags) [reply src/Haskoin/Address.hs:6]: https://github.com/haskell-bitcoin/bitcoin/issues/14

File: src/Haskoin/Address.hs (hunk 3 above)
Comment (ProofOfKeags) [reply src/Haskoin/Address.hs]: Eh taking care of it now

Comment (ProofOfKeags) [reply src/Haskoin/Address.hs]: OK this is done for the modules that were actually touched as part of this PR. subsequent ones can be done later.

File: fourmolu.yaml (hunk 8 above)
Comment (ProofOfKeags) [reply fourmolu.yaml:1]: we're doing 4 spaces normally and 2 space wheres. let it be so.

File: CHANGELOG.md [hunk 9]
```diff
@@ -1,315 +1,12 @@
 # Changelog
+
 All notable changes to this project will be documented in this file.
 
 The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
```
Comment (GambolingPangolin) [inline CHANGELOG.md:6]: We should probably make a note of the version of `haskoin-core` that we forked.  The diligent can find it in the git history but IMO it's good to note it here too.

File: README.md [hunk 10]
```diff
@@ -17,4 +16,4 @@ Haskoin Core is a library of Bitcoin and Bitcoin Cash functions written in Haske
 
 ## Contributing
 
-Please use `ormolu` (or `fourmolu`) to format code prior to submission.  See `scripts/pre-commit.sh` for an example pre-commit hook.
+Please use `ormolu` (or `fourmolu`) to format code prior to submission. See `scripts/pre-commit.sh` for an example pre-commit hook.
```
Comment (GambolingPangolin) [inline README.md]: We should insist on `fourmolu` since we are including a configuration file for it and using 4 spaces.

File: package.yaml [hunk 11]
```diff
@@ -1,14 +1,15 @@
-name: haskoin-core
-version: 0.21.2
+name: bitcoin
+version: 0.0.0
```
Comment (GambolingPangolin) [inline package.yaml]: I would prefer to start on `0.1.0` if it's all the same to you.

File: package.yaml [hunk 12]
```diff
@@ -1,14 +1,15 @@
...
 synopsis: Bitcoin & Bitcoin Cash library for Haskell
```
Comment (GambolingPangolin) [inline package.yaml]: Not a Bitcoin Cash library.  Prob should grep for "bitcoin cash" to make sure we get everything.

File: README.md [hunk 13]
```diff
@@ -4,7 +4,6 @@ Haskoin Core is a library of Bitcoin and Bitcoin Cash functions written in Haske
```
Comment (GambolingPangolin) [inline README.md:4]: No Bitcoin Cash functions.  Also we should probably change the title of the readme to e.g. "Haskell Bitcoin".

File: src/Haskoin/Address.hs [hunk 14]
```diff
@@ -145,68 +140,69 @@ instance Serial Address where
...
-addrToText net a@PubKeyAddress{getAddrHash160 = h}
-    | isNothing (getCashAddrPrefix net) =
-        Just . encodeBase58Check . runPutS $ base58put net a
-    | otherwise = cashAddrEncode net 0 (runPutS $ serialize h)
-addrToText net a@ScriptAddress{getAddrHash160 = h}
-    | isNothing (getCashAddrPrefix net) =
-        Just . encodeBase58Check . runPutS $ base58put net a
-    | otherwise =
-        cashAddrEncode net 1 (runPutS $ serialize h)
+addrToText net a@PubKeyAddress{getAddrHash160 = h} = Just . encodeBase58Check . runPutS $ base58put net a
```
Comment (GambolingPangolin) [inline src/Haskoin/Address.hs]: The pattern bindings are unused here and in the next definition.

**Comment by ProofOfKeags** [comment]
all comments addressed

File: package.yaml [hunk 15]
```diff
@@ -1,14 +1,15 @@
-name: haskoin-core
-version: 0.21.2
-synopsis: Bitcoin & Bitcoin Cash library for Haskell
+name: bitcoin
+version: 0.1.0
+synopsis: Bitcoin library for Haskell
 description: Please see the README on GitHub at <https://github.com/haskoin/haskoin-core#readme>
```
Comment (GambolingPangolin) [inline package.yaml]: The URLs need to be updated (here and a few below)

Comment (ProofOfKeags) [reply package.yaml]: ah nice good catch

**Review by GambolingPangolin — APPROVED** [review]

**Pull request merged**